cowprotocol/services
# Consider token as good when node doesn't support trace calls

# Description
Not all of our fallback nodes support the `trace_call` API. While we can configure the tracing URL separately, in case of outage the API may not be able to classify any tokens as good and thus effectively prevent trading.

# Changes
- [x] If trace call many fails due to a TransportError (which is what happens if you call an unsupported method), consider the token as good.
- [x] Add tracing spans with token address context for any logs that happen during detection

## How to test
```
cargo test --package shared --lib -- bad_token::trace_call::tests::mainnet_tokens --exact --show-output 
--ignored
```

both with a Node that does and does not support the trace call API. Observe that in the case of non-support all tokens are of quality good.
